movieReviewApp/src/main/java/com/mobiquity/movieReviewApp/service/PasswordRecoverServiceImpl.java

```
@@ -43,22 +47,29 @@ public String passwordActivationLink(String emailId) {
 
   @Override
   @Transactional
-  public String UpdatePassword(ResetPassword resetPassword) {
-    updateHashedPassword(resetPassword);
+  public String UpdatePassword(ForgotPassword forgotPassword) {
+    updateHashedPassword(forgotPassword.getPassword(),
+        getEmailIdForNewPassword(forgotPassword.getToken()));
     return "New Password is Updated";
   }
 
   @Override
   public String getEmailIdForNewPassword(String token) {
-    claim = utilityService.retrieveDataFromClaim(token);
-    return claim.getSubject().split(" ")[0];
+    try {
+      claim = utilityService.retrieveDataFromClaim(token);
+      return claim.getSubject().split(" ")[0];
+    } catch (ExpiredJwtException e) {
```

instead of handling here you can throw and handle in the controllerAdvice

---

No.its a runtime error not a error we throw so we have to catch it and throw it along with message.we can throw the same error with message but same is done with custom exception